reconcile/utils/jira_client.py

```diff
@@ -17,13 +18,25 @@
 from reconcile.utils.secret_reader import SecretReader
 
 
+class JiraWatcherSettings(Protocol):
+    read_timeout: int
+    connect_timeout: int
+
+
 class JiraClient:
     """Wrapper around Jira client."""
 
     def __init__(
-        self, jira_board: Mapping[str, Any], settings: Optional[Mapping] = None
+        self,
+        jira_board: Mapping[str, Any],
+        settings: Optional[Mapping] = None,
+        jira_watcher_settings: Optional[JiraWatcherSettings] = None,
+        secret_reader: Optional[SecretReader] = None,
     ):
-        self.secret_reader = SecretReader(settings=settings)
+        if not (secret_reader or settings):
+            raise RuntimeError("JiraClient needs secret_reader or settings.")
+        if not secret_reader:
+            self.secret_reader = SecretReader(settings=settings)
```

Comment:
Do you mean by assign `secret_reader` to `self.secret_reader` if passed else new.

```python
self.secret_reader = secret_reader if secret_reader is not None else SecretReader(settings=settings)
```

Reply:
This was actually a bug because I forgot to set the secret_reader if its given as parameter. Fixed now. Good catch!